cmd/kube-scheduler/app/server.go

```diff
@@ -79,6 +80,15 @@ kube-scheduler is the reference implementation.
 See [scheduling](https://kubernetes.io/docs/concepts/scheduling-eviction/)
 for more information about scheduling and the kube-scheduler component.`,
 		Run: func(cmd *cobra.Command, args []string) {
+			// Due to the default values may be changed by the feature gates,
+			// we need to update the ComponentConfig after the CLI flags are
+			// parsed.
+			cfg, err := latest.Default()
```

Comment:
We may need to ensure the feature gates get always parsed firstly. If you don't mind, I opened #103468 to fix this issue from another angle.